CompilerSource/parsing/tokens.h

```diff
@@ -0,0 +1,150 @@
+  TT_DECIMAL,         // A dot. '.'. Whether for number or for access is determined in the second pass.
+  TT_PLUS,            // The + operator.
+  TT_MINUS,           // The - operator.
+  TT_STAR,            // The * operator.
+  TT_SLASH,           // The / operator.
+  TT_PERCENT,         // The % operator.
+  TT_AMPERSAND,       // The & operator.
+  TT_PIPE,            // The | operator.
+  TT_CARET,           // The ^ operator.
+  TT_AND,             // The and/&& operator.
+  TT_OR,              // The or/|| operator.
+  TT_XOR,             // The xor/^^ operator.
+  TT_DIV,             // The div operator.
+  TT_MOD,             // The mod (modulo) operator.
+  TT_NOT,             // The not operator. This refers to the keyword, not the bang symbol (!).
+  TT_EQUALTO,         // The == operator. Always checks equality.
+  TT_NOTEQUAL,        // The != or <> operator.
+  TT_BANG,            // The ! operator.
+  TT_TILDE,           // The ~ operator.
+  TT_INCREMENT,       // The ++ operator.
+  TT_DECREMENT,       // The -- operator.
+  TT_LESS,            // The < operator (or opening angle bracket).
+  TT_GREATER,         // The > operator (or closing angle bracket).
+  TT_LESSEQUAL,       // The <= operator.
+  TT_GREATEREQUAL,    // The >= operator.
+  TT_LSH,             // The << operator.
+  TT_RSH,             // The >> operator (or double closing angle bracket).
+  TT_QMARK,           // ?
+  TT_BEGINPARENTH,    // (
+  TT_ENDPARENTH,      // )
+  TT_BEGINBRACKET,    // [
+  TT_ENDBRACKET,      // ]
+  TT_BEGINBRACE,      // {
+  TT_ENDBRACE,        // }
+  TT_BEGINTRIANGLE,   // <
+  TT_ENDTRIANGLE,     // >
+  TT_DECLITERAL,      // 0 1 2... 9 10 11... 9876543210...
+  TT_BINLITERAL,      // 0b0 0b1 0b10 0b11 0b100 0b101 0b110 ...
+  TT_OCTLITERAL,      // 0o0 0o1 0o2... 0o6 0o7 0o10 0o11... 0o76543210 ...
+  TT_HEXLITERAL,      // 0x0 $1 0x2... 0x9 0xA... $F $10... $FEDCBA9876543210...
+  TT_STRING,          // "", ''
+  TT_SCOPEACCESS,     // ::
+  TT_FUNCTION,        // game_end
+  TT_TYPE_NAME,       // int, double, whatever
+  TT_NAMESPACE,       // std, enigma, sf
+  TT_LOCGLOBAL,       // global/local
+  TT_SHORTSTATEMENT,  // Short statements; return, mostly
+  TT_TINYSTATEMENT,   // break, continue, exit...
+  TT_S_SWITCH,        // switch
+  TT_S_REPEAT,        // repeat
+  TT_S_CASE,          // case
+  TT_S_DEFAULT,       // default
+  TT_S_FOR,           // for
+  TT_S_IF,            // if
+  TT_S_ELSE,          // else
+  TT_S_DO,            // do
+  TT_S_WHILE,         // while
+  TT_S_UNTIL,         // while
+  TT_S_WITH,          // with
+  TT_S_TRY,           // try
+  TT_S_CATCH,         // catch
+  TT_S_NEW,           // new
+
+  TTM_WHITESPACE,     // Preprocessing token: whitespace
+  TTM_CONCAT,         // Preprocessing token: ##
+  TTM_STRINGIFY,      // Preprocessing token: #
+
+  TT_ERROR,
+  TT_ENDOFCODE
+};
+
+namespace internal {
+namespace useless {
```

Comment:
Ok but why is it useless? perhaps leave a reason for why its here so we can phase it out later